catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -304,6 +358,11 @@ public UpdateResponse update(UpdateRequest updateRequest) throws IngestException
                         "Found more metacards than updated metacards provided. Please ensure your attribute values match unique records.");
             }
 
+        } else if (pendingNrtIndex.getAllPresent(identifiers).size() > 0) {
+            Map<String, Metacard> matches = pendingNrtIndex.getAllPresent(identifiers);
+            for (Entry<String, Metacard> match : matches.entrySet()) {
```

Comment:
I tried this but it caused other tests to fail but it looks like they are failing since they are mutating the reference stored in the pending NRT index.  This worked after I added a defensive copy which now makes the pending NRT index safer.